addons/mail/static/src/new/core/thread_model.js

```diff
@@ -12,6 +12,15 @@ export class Thread {
     /** @type {import("@mail/new/core/messaging").Messaging['state']} */
     _state;
 
+    /**
+     * @param {Object} param0
+     * @param {string} param0.model
+     * @param {number} param0.id
+     */
+    static createLocalId({ model, id }) {
```

Comment:
Using a destruct here seams useless since you never just pass an object to the method.
Therefor I think you should just use a function with 2 params. It's also better to avoid miss usage of this function by passing a random object.

Reply:
I opted for an object rather than a bunch of parameters to avoid errors due to the inversion of the two arguments

Reply:
Proper tooling will tell you the arguments you need to fill